avro-converter/src/test/java/io/confluent/connect/avro/AvroDataTest.java

```diff
@@ -1248,4 +1250,52 @@ private void checkNonRecordConversionNull(Schema schema)
     Object converted = avroData.fromConnectData(schema, null);
     assertNull(converted);
   }
+
+  @Test
+  public void testToConnectDataStringEnumUnion() {
+    org.apache.avro.Schema enumSchema = org.apache.avro.SchemaBuilder.enumeration("TestEnum").symbols("A", "B", "C", "D");
+
+    Map<org.apache.avro.Schema, Object> avroSchemaToValueTests = new LinkedHashMap<>();
+    avroSchemaToValueTests.put(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.STRING), "AnotherValue");
+    avroSchemaToValueTests.put(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.INT), 1000);
+    avroSchemaToValueTests.put(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.LONG), 1000L);
+    avroSchemaToValueTests.put(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.BOOLEAN), false);
+    avroSchemaToValueTests.put(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.DOUBLE), Double.MAX_VALUE);
+    avroSchemaToValueTests.put(org.apache.avro.Schema.create(org.apache.avro.Schema.Type.FLOAT), Float.MAX_VALUE);
+
+    for(Map.Entry<org.apache.avro.Schema, Object> kvp:avroSchemaToValueTests.entrySet()){
+      org.apache.avro.Schema unionSchema = org.apache.avro.SchemaBuilder.unionOf()
+          .type(kvp.getKey())
+          .and()
+          .type(enumSchema)
+          .endUnion();
+      GenericData.EnumSymbol enumValue = (GenericData.EnumSymbol) GenericData.get().createEnum("A", enumSchema);
+      assertTrue(
```

Comment:
these assertions are a bit confusing... validate() returns true if validation succeeds but from what you seem to be testing here, it should be false?